### ISSUE TITLE
Status code was 410 and not [204]: HTTP Error 410: Gone

### ISSUE THREAD

**Issue by zviederi** [ansible-ThoTeam/nexus3-oss#264]
Hi,
I got the following error for latest Nexus/3.21.2-03:

```
> fatal: [dev0]: FAILED! => {
>     "changed": false,
>     "connection": "close",
>     "content": "{\n  \"name\" : \"create_blobstores_from_list\",\n  \"result\" : \"Creating and updating scripts is disabled\"\n}",
>     "content_length": "102",
>     "content_type": "application/json",
>     "date": "Mon, 23 Mar 2020 21:07:16 GMT",
>     "elapsed": 0,
>     "invocation": {
>         "module_args": {
>             "attributes": null,
>             "backup": null,
>             "body": {
>                 "content": "import groovy.json.JsonOutput\nimport groovy.json.JsonSlurper\n\nparsed_args = new JsonSlurper().parseText(args)\n\nList<Map<String, String>> actionDetails = []\nMap scriptResults = [changed: false, error: false]\nscriptResults.put('action_details', actionDetails)\n\nparsed_args.each { blobstoreDef ->\n\n    Map<String, String> currentResult = [name: blobstoreDef.name, type: blobstoreDef.get('type', 'file')]\n\n    existingBlobStore = blobStore.getBlobStoreManager().get(blobstoreDef.name)\n    if (existingBlobStore == null) {\n        try {\n            if (blobstoreDef.type == \"S3\") {\n                blobStore.createS3BlobStore(blobstoreDef.name, blobstoreDef.config)\n                msg = \"S3 blobstore {} created\"\n            } else {\n                blobStore.createFileBlobStore(blobstoreDef.name, blobstoreDef.path)\n                msg = \"File blobstore {} created\"\n            }\n            log.info(msg, blobstoreDef.name)\n            currentResult.put('status', 'created')\n            scriptResults['changed'] = true\n        } catch (Exception e) {\n            log.error('Could not create blobstore {}: {}', blobstoreDef.name, e.toString())\n            currentResult.put('status', 'error')\n
>            scriptResults['error'] = true\n            currentResult.put('error_msg', e.toString())\n        }\n    } else {\n        log.info(\"Blobstore {} already exists. Left untouched\", blobstoreDef.name)\n        currentResult.put('status', 'exists')\n    }\n\n    scriptResults['action_details'].add(currentResult)\n}\n\nreturn JsonOutput.toJson(scriptResults)",
>                 "name": "create_blobstores_from_list",
>                 "type": "groovy"
>             },
>             "body_format": "json",
>             "client_cert": null,
>             "client_key": null,
>             "content": null,
>             "creates": null,
>             "delimiter": null,
>             "dest": null,
>             "directory_mode": null,
>             "follow": false,
>             "follow_redirects": "safe",
>             "force": false,
>             "force_basic_auth": true,
>             "group": null,
>             "headers": {
>                 "Content-Type": "application/json"
>             },
>             "http_agent": "ansible-httpget",
>             "method": "POST",
>             "mode": null,
>             "owner": null,
>             "password": "VALUE_SPECIFIED_IN_NO_LOG_PARAMETER",
>             "regexp": null,
>             "remote_src": null,
>             "removes": null,
>             "return_content": false,
>             "selevel": null,
>             "serole": null,
>             "setype": null,
>             "seuser": null,
>             "src": null,
>             "status_code": [
>                 "204"
>             ],
>             "timeout": 30,
>             "unix_socket": null,
>             "unsafe_writes": null,
>             "url": "http://dev0:9080/nexus/service/rest/v1/script",
>             "url_password": "VALUE_SPECIFIED_IN_NO_LOG_PARAMETER",
>             "url_username": "admin",
>             "use_proxy": true,
>             "user": "admin",
>             "validate_certs": false
>         }
>     },
>     "json": {
>         "name": "create_blobstores_from_list",
>         "result": "Creating and updating scripts is disabled"
>     },
>     "msg": "Status code was 410 and not [204]: HTTP Error 410: Gone",
>     "redirected": false,
>     "server": "Nexus/3.21.2-03 (OSS)",
>     "status": 410,
>     "url": "http://dev0:9080/nexus/service/rest/v1/script",
>     "x_content_type_options": "nosniff"
> }
```
Part of playbook:
```
    ...
    nexus_timezone: 'Europe/Riga'
    nexus_upgrade: true
    nexus_default_port: 9080
    nexus_api_hostname: dev0
    nexus_default_context_path: '/nexus/'
    nexus_admin_password: "{{vault_nexus_password}}"
    nexus_delete_default_repos: true
    nexus_delete_default_blobstore: true
    nexus_anonymous_access: false
    nexus_backup_dir: '/var/nexus-backup'

    nexus_blobstores:
      - name: docker
        path: "{{nexus_data_dir}}/blobs/docker"
      - name: maven
        path: "{{nexus_data_dir}}/blobs/maven"
    ....
```
I tried several times on pure target ubuntu 16.04 host.
Any ideas what is the cause for this error? Need some additional configs in playbook?

**Comment by zeitounator** [comment]
This is most probably due to groovy being disabled by default. I'm working on a temporary patch for now

**Comment by zeitounator** [comment]
Fix is currently going into CI in #267 and will be merged soon. Please note that this will enable groovy script creation as it was the case prior to version 3.21.2 but is now turned off by default. Since the role massively relies on groovy script for historical reasons, I don't have the choice until I can move all provisioning to REST API calls.
